### DISCUSSION TITLE
How to repeatedly run a simulation for a single year

### DISCUSSION THREAD

**Discussion by zi-hua-liu** [NOAA-GFDL/CEFI-regional-MOM6#93] [Q&A]
Hi All,

I am trying to repeatedly run a simulation for the year 1990 and would appreciate some guidance on how to achieve this. Here’s what I’ve done so far:

- Modified the time variable in all input data files from 1990 to 1991.
- Changed 'current_date' in the "input.nml" file from '1990,1,1,0,0,0' to '1991,1,1,0,0,0'; with input_filename = 'r'.
- Updated the date in the "diag_table" from '1990  1  1' to '1991  1  1'.
- Restarted the model.

The model runs successfully, but the results are identical to the initial run for 1990, which is not what I expected. During the first run, there was no sea ice in January, but sea ice appeared in December. I was hoping to see some sea ice in January during the repeated runs.

However, I noticed that the 'Current model time' in "coupler.res" is 1990-12-31, not 1991-1-1, which probably could explain why it did not work as expected. I then tried setting the time in the input data files to 1990-12-31, but the model crashed with the fatal error:

"FATAL from PE 0: diag_manager_mod::register_diag_field: file=ice_month: Invalid date. Date=1991-02-31 00:00:00."

My question is: How can I run the simulation repeatedly for 3 or 4 years, using the last day of each year as the initial condition (i.e., restarting from the last day of the previous year) and still use input data for 1990 for each run?

I would appreciate any advice or suggestions. Thanks!

**Comment by zi-hua-liu** [comment]
I have solved this issue. :)

**Reply by theresa-cordero** [reply]
Hi Zihua, I'm glad you got it working! What did you end up changing about your setup?

**Reply by zi-hua-liu** [reply]
Thanks for checking with me. I have explored two approaches to manage the repeated years in the input data:

- Data concatenation:

I combined the input data for the repeated years into a single data file. However, this resulted in a very large file, which may not be ideal.

- Adding an extra time point:

Kept the input data for repeated years but added an additional time point (1991-01-01) to the 1990 data. This allows for a complete annual run, as without this addition, the monthly data output only includes 11 time series. With the extra time point, the model can restart from 1991-01-01 using the same input files but with updated time variables (from 1990 to 1991). 

Both methods are somewhat cumbersome and are related to correctly setting up the time variables. I would appreciate suggestions for more flexible and easier approaches if there are any!

**Reply by theresa-cordero** [reply]
I would suggest adding the `modulo = " " `  attribute to the time variable in the data file. With that option you shouldn't need to concatenate files or add an extra data point. 

You should be able to add that modulo attribute with this command:
`ncatted -a modulo,time,c,c," " filename.nc`
[see here for more detail](
https://stackoverflow.com/questions/58673282/how-to-make-time-variable-continuous-in-a-netcdf-forcing-climatology-file)

**Reply by zi-hua-liu** [reply]
Thanks very much! I see. Will add this to the time variable.
